packages/babel-plugin-proposal-partial-application/src/index.js

```diff
@@ -0,0 +1,192 @@
+import { declare } from "@babel/helper-plugin-utils";
+import syntaxPartialApplication from "@babel/plugin-syntax-partial-application";
+import { types as t } from "@babel/core";
+
+export default declare(api => {
+  api.assertVersion(7);
+
+  /**
+   * a function to figure out if a call expression has
+   * ArgumentPlaceholder as one of its arguments
+   * @param node a callExpression node
+   * @returns boolean
+   */
+  function hasArgumentPlaceholder(node) {
+    return node.arguments.some(arg => t.isArgumentPlaceholder(arg));
+  }
+
+  /**
+   * Unwrap the arguments of a CallExpression, and creates
+   * assignmentExpression from them and returns them as an array.
+   * @param {Object} node CallExpression node
+   * @returns {Array<AssignmentExpression>}
+   */
+  function unwrapArguments(node, scope) {
+    let newNode;
+    let argId;
+    return node.arguments.reduce((acc, arg) => {
+      if (!t.isArgumentPlaceholder(arg) && !t.isNumericLiteral(arg)) {
```

Comment:
Instead of `t.isNumericLiteral` you can use `t.isImmutable`, which also handles strings/booleans